.github/scripts/launch-ec2.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Check for AWS CLI and credentials
+if ! command -v aws &>/dev/null; then
+  echo "Error: AWS CLI is not installed. Please install it and configure your credentials."
+  exit 1
+fi
+
+if ! aws sts get-caller-identity &>/dev/null; then
+  echo "Error: Unable to access AWS. Ensure your credentials are configured correctly."
+  exit 1
+fi
+
+# Set variables
+AWS_REGION="us-east-2"
+# TODO document that this key needs to be created
+KEY_NAME="dandihub-gh-actions"
+# TODO create if DNE
+# allow gh-actions to ssh into ec2 job instance from anywhere
+SECURITY_GROUP_ID="sg-0bf2dc1c2ff9c122e"
+# TODO retrieve subnet id (public, created by dandi-hub eks-dandihub-public-us-east-2a)
+SUBNET_ID="subnet-0f544cca61ccd2804"
+AMI_ID="ami-088d38b423bff245f"
+EFS_ID="fs-02aac16c4c6c2dc27"
+LOCAL_SCRIPTS_DIR=".github/scripts"
+REMOTE_SCRIPTS_DIR="/home/ec2-user/scripts"
+MOUNT_POINT="/mnt/efs"
+ENV_FILE=".ec2-session.env"
```

Comment:
I dislike the fact that it would be just dumping into some hidden file in my current directory.
Could we make it dumped into some tmpdir and may be establish there env variable with its path so cleanup script could take it from that environment  

```suggestion
ENV_FILE="/run/user/$(id -u)/ec2-session.env"
```

but then we might want to add logic to react if file already exist -- since that would mean likely that cleanup did not remove it and instance might still be running etc.

Reply:
I liked having the .ec2-session.env right there so I could see if i had an instance running. So I've left as is--but if others end up using these  scripts, by all means feel free to change.